contracts/CheckBitcoinSigs.sol

```diff
@@ -16,26 +16,36 @@ library CheckBitcoinSigs {
     /// @dev             the address is the last 20 bytes of the keccak256 of the address
     /// @param _pubkey   the public key
     /// @return          the account address
-    function accountFromPubkey(bytes memory _pubkey) public pure returns (address) {
-        require(_pubkey.length == 64);
+    function accountFromPubkey(bytes memory _pubkey) internal pure returns (address) {
+        require(_pubkey.length == 64, "Pubkey must be 64-byte raw, uncompressed key.");
 
         // keccak hash of uncompressed unprefixed pubkey
-        bytes memory _digest = abi.encodePacked(keccak256(_pubkey));
-        return _digest.toAddress(_digest.length - 20);  // last 20 bytes
+        bytes32 _digest = keccak256(_pubkey);
+        return address(uint160(uint256(_digest)));
     }
 
     /// @notice          calculates the p2wpkh output script of a pubkey
     /// @dev             pads uncompressed pubkeys to 65 bytes as required by Bitcoin
```

Comment:
`converts an uncompressed pubkey (64-byte or 65-byte) to a compressed 33-byte pub key as required by Bitcoin`?